src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -51,15 +51,14 @@ public Product findProductByIdOrThrowException(UUID productId) {
     }
 
     @Override
-    public void deleteProduct(UUID productId){
+    public ProductResponse deleteProduct(UUID productId){
         ifProductIsNotDeletedThrowException(productId);
         Product product = findProductByIdOrThrowException(productId);
         product.setDeleted(true);
-        productRepository.save(product);
+        return ProductMapper.toProductResponse(productRepository.save(product));
```

Comment:
Si ponés el delete como void no haría falta devolver el product response por lo que esto debería quedarte así
```suggestion
        productRepository.save(product);
```